backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java

```diff
@@ -83,6 +83,15 @@ public BucketDto findBucketByName(String name) {
     return modelMapper.map(bucketEntity, BucketDto.class);
   }
 
+  @Override
+  public List<BucketResponse> listAllDeletedBuckets() {
+    List<BucketEntity> entityList = bucketRepository.findAllByDeletedOnIsNotNull();
```

Comment:
You can make` bucketRepository.findAllByDeletedOnIsNotNull()` return Set instead of List, JPA allows that by default , so you won't have conversion in next line